source/tutorials/hpc.rst

```diff
@@ -0,0 +1,693 @@
+         
+         sudo mkdir /etc/munge
+         dd if=/dev/urandom bs=1 count=1024 | sudo tee -a /etc/munge/munge.key
+
+   #. Set the ownership to `munge` and set the correct access permissions.
+     
+      .. code-block:: bash
+         
+         sudo chown munge: /etc/munge/munge.key
+         sudo chmod 400 /etc/munge/munge.key      
+
+   #. Start the munge service and set it to start automatically on boot.
+
+      .. code-block:: bash
+               
+         sudo systemctl enable munge --now
+
+#. Test munge.
+
+   a. Create a munge credential.
+
+      .. code-block:: bash
+
+         munge -n
+
+      Example output:
+
+      .. code-block:: console
+
+         MUNGE:AwQFAAC8QZHhL/+Fqhalhi+ZJBD5LavtMa8RMles1aPq7yuIZq3LtMmrB7KQZcQjG0qkFmoIIvixaCACFe1stLmF4VIg4Bg/7tilxteXHS940cuZ/TxpIuqC6fUH8zLgUZUPwJ4=:
+
+
+   #. Validate a munge credential.  
+      
+      .. code-block:: bash
+
+         munge -n | unmunge | grep STATUS
+
+      Example output:
+        
+      .. code-block:: console
+
+         STATUS:           Success (0)
+
+Set up worker nodes
+*******************
+
+For each worker node, perform these steps:
+
+#. Install |CL| on the worker node and add a user with adminstrator privilege.  
+   Ensure the username is the same as the one on the controller node.  This
+   is needed to simplify password-less-SSH-access setup in the next section.  
+
+#. Boot it up and log in.
+
+#. Update |CL| to the latest version.
+
+   .. code-block:: bash
+
+      sudo swupd update
+
+#. Set the password for the root user.
+
+   .. code-block:: bash
+
+      sudo passwd root
+
+#. Permit the root user to SSH.  This is necessary for password-less SSH setup 
+   later.
+
+   .. code-block:: bash
+
+      echo "PermitRootLogin yes" | sudo tee -a /etc/ssh/sshd_config 
+      sudo systemctl restart sshd
+
+#. Set the hostname to `hpc-worker` plus its number.
+
+   .. code-block:: bash
+
+      sudo hostnamectl set-hostname hpc-worker<number>
+
+#. Set the date and time to synchronize with an NTP server.
+
+   .. code-block:: bash
+
+      sudo timedatectl set-ntp true
+
+#. Install the `cluster-tools` bundle.
+
+   .. code-block:: bash
+
+      sudo swupd bundle-add cluster-tools
+
+Set up password-less SSH access and PDSH on all nodes
+*****************************************************
+
+To efficiently manage a cluster, it is useful to have a tool
+that allows issuing the same command to multiple nodes at once.  
+And that tool is :abbr:`PDSH (Parallel Distributed Shell)`, which is included
+with the `cluster-tools` bundle. PDSH is built with slurm support, so it can 
+access hosts as defined in the slurm partitions.  PDSH relies on password-less 
+SSH access in order for it to work properly.  
+
+#. Log into the controller node with the regular user account.
+
+#. Generate an SSH authentication key pair for the regular user.
+
+   .. code-block:: bash
+
+      ssh-keygen -t rsa
+
+   Accept all default options.
+
```

Comment:
Maybe stress the fact that in particular we don't want any passphrase (it should be the default)

Reply:
done.